providers/inline-evaluating-provider/main/java/dev/openfeature/contrib/providers/inlineeval/RuleFetcher.java

```diff
@@ -0,0 +1,20 @@
+package dev.openfeature.contrib.providers.inlineeval;
+
+import dev.openfeature.sdk.EvaluationContext;
+
+public interface RuleFetcher {
+
+    /**
+     * Called to set up the client initially. This is used to pre-fetch initial data as well as setup mechanisms to stay up
+     * to date.
+     * @param initialContext application context known thus far
+     */
+    void initialize(EvaluationContext initialContext);
+
+    /**
+     * Given a key name, return the JSONLogic rules for it
+     * @param key The key to fetch logic for
+     * @return json logic rules or null
+     */
+    String getRuleForKey(String key);
+}
```

Comment:
Abstracting this into an interface seems like a good idea. As a thought experiment, I imagined adding a simple additional RPC to flagd to send it's ruleset when it's called. In that hypothetical flow, the ruleset would be cached with `initialize` and then `getRuleForKey` could be used for evaluation. This doesn't QUITE work for flagd, since in the flagd schema the result of the rule is actually a `variant` which is the [key representing the value](https://github.com/open-feature/flagd/blob/main/config/samples/example_flags.flagd.json#L47-L70). Some additional mapping would be required to pull the associated value out of flagd's variant.

Reply:
We could do that, sure. It would read json from a file and do a simple get by key.